spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelAutoConfiguration.java

```diff
@@ -76,23 +80,47 @@
 
 	@Bean
 	@ConditionalOnMissingBean
-	OpenTelemetry otel(TracerProvider tracerProvider, ContextPropagators contextPropagators) {
-		OpenTelemetry openTelemetry = DefaultOpenTelemetry.builder().setTracerProvider(tracerProvider)
+	OpenTelemetry otel(SdkTracerProvider tracerProvider, ContextPropagators contextPropagators) {
+		// this is super hacky. it would be better to go and eliminate all usage of the
+		// global. That requires some changes upstream in the instrumentation APIs, so for
+		// now, we hack.
+		GlobalOpenTelemetry.resetForTest();
```

Comment:
as of 0.15.0, if you try to `set` more than once on the Global, an exception is thrown. This is to prevent initialization conflicts. The only reason this needs to be here to support integration testing. If there's an approach that could be applied to the testing that would remove the need, that would be excellent. In the meantime, I'm going to work on making the instrumentation API so that it doesn't use the global, but can have the instance injected into it (note: currently, the issue is with the use of the Propagators on the global instance...the Tracer is already injected).

Reply:
I don't think it's a good idea to have this here. Can't we call that method for tests only? We could create a test execution listener that is executed before each test and calls that static method.

Reply:
otel agent just published 0.15.1 last night and I think we might have a fix for this in that patch release. I'll take a look later today.

Reply:
looks like the fix didn't make it into 0.15.1, unfortunately. it'll be there in 0.16.0

Reply:
> I don't think it's a good idea to have this here. Can't we call that method for tests only? We could create a test execution listener that is executed before each test and calls that static method.

This sounds like a good idea. Do you have an example I can crib from?

Reply:
Sure - https://docs.spring.io/spring-framework/docs/current/reference/html/testing.html#spring-testing-annotation-testexecutionlisteners

Example:

```java
package a.b.c;

public class OTelTestExecutionListener implements TestExecutionListener, Ordered {
    
    public void beforeTestClass(TestContext testContext) throws Exception {
        GlobalOpenTelemetry.resetForTest();
    }; 

    @Override
    public int getOrder() {
        return Integer.MAX_VALUE;
    };
}
```

then in a file `src/test/resources/META-INF/spring.factories` register the extension:

```properties
org.springframework.test.context.TestExecutionListener=\
a.b.c.OTelTestExecutionListener
```

Reply:
With the 0.16.0 release, we no longer need to do any of these shenanigans, so this is all unnecessary.

Reply:
Just realized I forgot to delete this line.. I'll get it fixed up. sorry about that!

Reply:
done!